File: Week11Ex/Week11_BoundService/app/src/main/java/com/example/week10_boundservice/MainActivity.java
package com.example.week10_boundservice;

import androidx.appcompat.app.AppCompatActivity;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.content.ServiceConnection;
import android.os.Bundle;
import android.os.IBinder;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    BoundService boundService = new BoundService();
    boolean isConnected = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Intent intent = new Intent(this, BoundService.class);
        bindService(intent, serviceConnection, Context.BIND_AUTO_CREATE);
    }

    public void getTime(View v) {
        TextView textView = findViewById(R.id.time);
        textView.setText(BoundService.getSysTime());
    }

    private ServiceConnection serviceConnection = new ServiceConnection() {
        @Override
        public void onServiceConnected(ComponentName componentName, IBinder service) {
            BoundService.MyLocalBinder binder = (BoundService.MyLocalBinder) service;
            boundService = binder.getBoundService();
            isConnected = true;
        }

        @Override
        public void onServiceDisconnected(ComponentName componentName) {
            isConnected = false;
        }
    };
}

File: Week9Ex/Week9/app/src/main/java/com/example/week9/MainActivity.kt
package com.example.week9

import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Switch
import android.widget.TextView
import kotlin.properties.Delegates

class MainActivity : AppCompatActivity() {
    private lateinit var textView:TextView
    private lateinit var editText: EditText
    private lateinit var applyBtn:Button
    private lateinit var saveBtn:Button
    private lateinit var switch: Switch

    private lateinit var text:String
    private var switchOnOff:Boolean = false

    val SHARED_PREFS = "sharedPrefs"
    val TEXT = "text"
    val SWITCH1 = "switch1"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        textView = findViewById(R.id.textView)
        editText = findViewById(R.id.editText)
        applyBtn = findViewById(R.id.applyBtn)
        saveBtn = findViewById(R.id.saveBtn)
        switch = findViewById(R.id.switchBtn)

        applyBtn.setOnClickListener(View.OnClickListener {
            textView.text = editText.text.toString()
        })

        saveBtn.setOnClickListener(View.OnClickListener {
            saveData()
        })

        loadData()
        updateViews()
    }

    private fun saveData() {
        val sharedPreferences:SharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE)
        val editor:SharedPreferences.Editor = sharedPreferences.edit()
        editor.putString(TEXT, textView.text.toString())
        editor.putBoolean(SWITCH1, switch.isChecked)
        editor.apply();
    }

    private fun loadData() {
        val sharedPreferences:SharedPreferences = getSharedPreferences(SHARED_PREFS, MODE_PRIVATE)
        text = sharedPreferences.getString(TEXT, "").toString()
        switchOnOff = sharedPreferences.getBoolean(SWITCH1, false)
    }

    private fun updateViews() {
        textView.text = text
        switch.isChecked = switchOnOff
    }
}